### DISCUSSION TITLE
How to set app@sharepoint as taxonomy administrator in the modern termstore

### DISCUSSION THREAD

**Discussion by jaspervoskuilen** [pnp/pnpcore#1130] [Q&A] [answered]
As stated in the [docs](https://pnp.github.io/pnpcore/using-the-sdk/taxonomy-intro.html)  and in an older [issue](https://github.com/SharePoint/sp-dev-docs/issues/6155), the account app@sharepoint needs to be added as a taxonomy administrator. That used to be easy enough in the classic experience.

However the tenant I needed this for had the classic termstore experience removed/phased out. All that is left in the tenant admin section is the modern termstore experience, which is nice, but doesn't let me add the needed account.

How can I add this account (i:0i.t|00000003-0000-0ff1-ce00-000000000000|app@sharepoint ) through the modern experience or through other methods?

**Comment by jansenbe** [comment, marked as answer]
Can't you reach https://contoso-admin.sharepoint.com/_layouts/15/TermStoreManager.aspx? Update contoso with your tanant name

**Reply by jaspervoskuilen** [reply]
No that page redirects back to the new modern termstore management page.

If I look in the advanced functions, then it mentions the classic page has been removed a few weeks back.

**Reply by jansenbe** [reply]
Interesting, for my tenant the page will be removed in April. There currently is not Graph or CSOM API for adding term store admins, using Fiddler on the modern admin center site shows the used API, question is whether this one still accepts the app@sharepoint as input?

Do you have bandwidth to give this a try?

![image](https://user-images.githubusercontent.com/7451219/223360156-8dc2a4dd-277c-4994-865b-165e51734e9a.png)

**Reply by jaspervoskuilen** [reply]
That worked!
It replaces any previously set up managers, but at least I end up with the app@sharepoint account
I used Pnp PowerShell to make the call like this:
```
$body = "{`"administrators`":[{`"user`": { `"email`": `"app@sharepoint`", `"userPrincipalName`": `"i:0i.t|00000003-0000-0ff1-ce00-000000000000|app@sharepoint`", displayName: `"SharePoint App`" }}]}"
Invoke-PnPSPRestMethod -Method PATCH -Url "/_api/v2.1/termStore?select=*,administrators" -Raw -Content $body
```

**Reply by jansenbe** [reply]
Thanks for testing this and providing feedback @jaspervoskuilen . I'll add this approach to our docs.
